src/server/node_manager.c

```diff
@@ -2715,6 +2715,13 @@ discard_job(job *pjob, char *txt, int noack)
 	int	 rc;
 	int	 rver;
 
+	/* We're about to discard the job, reply to a preemption. 
+	 * This serves as a catch all just incase the code doesn't reply on its own.
+	 */
+
+	if (pjob->ji_pmt_preq != NULL)
+		reply_preempt_jobs_request(PBSE_NONE, PREEMPT_METHOD_DELETE, pjob);
+
```

Comment:
We won't call this function twice because it NULLs ji_pmt_preq.  If we come across another time we could call the function, ji_pmt_preq will be NULL, so we won't call it.